Ported kinesis exporter from Omnition

owais opened the pull request:
Porting the existing kinesis exporter from Omnition's Otel distribution
to contrib.

Porting from: https://github.com/Omnition/omnition-opentelemetry-collector/tree/master/exporter/kinesis

Currently exports to Kinesis in Jaeger proto format only. Support for other formats can be added later.

owais commented:
Looking at CI failures.

tigrannajaryan commented on go.mod:
```diff
@@ -22,3 +24,5 @@ require (
 replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azuremonitorexporter v0.0.0 => ./exporter/azuremonitorexporter
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sapmexporter => ./exporter/sapmexporter
```
Can you remove these "replace" directives that are no longer needed?

owais replied:
I can but first we need to merge this or we'll need another PR I think. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/59

pjanotti replied:
I kind of prefer to go in two PRs but up to you.

tigrannajaryan reviewed (approved):
I had reviewed this code earlier in its original repo, approving.

pjanotti commented on exporter/kinesisexporter/config_test.go:
```diff
@@ -0,0 +1,119 @@
+
+func TestConfigCheck(t *testing.T) {
+	cfg := (&Factory{}).CreateDefaultConfig()
+	assert.NoError(t, configcheck.ValidateConfig(cfg))
```
👍

pjanotti commented on exporter/kinesisexporter/config.go:
```diff
@@ -0,0 +1,55 @@
+type KPLConfig struct {
+	AggregateBatchCount  int `mapstructure:"aggregate_batch_count"`
+	AggregateBatchSize   int `mapstructure:"aggregate_batch_size"`
+	BatchSize            int `mapstructure:"batch_size, omitempty"`
```
nit: remove `omitempty`

pjanotti reviewed (approved):
It will be nice to comment on the configuration fields but due to the number of them let's do it separately. I also had looked at this code before but tried to look again... single nit, LGTM.

owais committed:
Ported kinesis exporter from Omnition

Porting the existing kinesis exporter from Omnition's Otel distribution
to contrib.

Porting from: https://github.com/Omnition/omnition-opentelemetry-collector/tree/master/exporter/kinesis

owais commented:
Updated.
